### PR TITLE
fix prediction-request-rag

### PR DESCRIPTION
**Pull request by moarshy**
this fixes errors encountered in mech over the last week.

### CODE REVIEW

File: packages/napthaai/customs/prediction_request_rag/prediction_request_rag.py [hunk 1]
```diff
@@ -437,22 +478,45 @@ def fetch_additional_information(
...
     docs = [doc for doc in docs if doc]
 
+    # remove empty documents ""
+    filtered_docs = []
+    for doc in docs:
+        try:
+            if doc.text != "":
+                filtered_docs.append(doc)
+        except Exception as e:
+            continue
```
Comment (Adamantios) [inline packages/napthaai/customs/prediction_request_rag/prediction_request_rag.py:488]: What exception could occur when trying to append here?

File: packages/napthaai/customs/prediction_request_rag/prediction_request_rag.py [hunk 2]
```diff
@@ -437,22 +478,45 @@ def fetch_additional_information(
...
     split_docs = [doc for doc in split_docs if doc]
 
+    # remove empty documents ""
+    filtered_split_docs = []
+    for doc in split_docs:
+        try:
+            if doc.text != "":
+                filtered_split_docs.append(doc)
+        except Exception as e:
+            continue
```
Comment (Adamantios) [inline packages/napthaai/customs/prediction_request_rag/prediction_request_rag.py:515]: Same question.

File: packages/napthaai/customs/prediction_request_rag/prediction_request_rag.py (hunk 1 above)
Comment (moarshy) [reply packages/napthaai/customs/prediction_request_rag/prediction_request_rag.py:488]: I was getting an error when accessing `doc.text` in some cases. wasn't able to identify why

File: packages/napthaai/customs/prediction_request_rag/prediction_request_rag.py (hunk 2 above)
Comment (moarshy) [reply packages/napthaai/customs/prediction_request_rag/prediction_request_rag.py:515]: will do without try/except

File: packages/napthaai/customs/prediction_request_rag/prediction_request_rag.py (hunk 1 above)
Comment (Adamantios) [reply packages/napthaai/customs/prediction_request_rag/prediction_request_rag.py:488]: So how about:

```python
for doc in docs:
    try:
        text = doc.text
    except {{Add the specific exception here}}:
        # also some message here maybe?
        continue
    if text:
        filtered_docs.append(doc)
```

File: packages/napthaai/customs/prediction_request_rag/prediction_request_rag.py (hunk 2 above)
Comment (Adamantios) [reply packages/napthaai/customs/prediction_request_rag/prediction_request_rag.py:515]: Well this is accessing `text` too (https://github.com/valory-xyz/mech/pull/184#discussion_r1510933808).

File: packages/napthaai/customs/prediction_request_rag/prediction_request_rag.py (hunk 1 above)
Comment (moarshy) [reply packages/napthaai/customs/prediction_request_rag/prediction_request_rag.py:488]: `filtered_docs = [doc for doc in docs if hasattr(doc, 'text') and doc.text != ""]`

maybe can do like this?

Comment (Adamantios) [reply packages/napthaai/customs/prediction_request_rag/prediction_request_rag.py:488]: Yes, if the exception was about the attribute not existing that would be much better.

Comment (moarshy) [reply packages/napthaai/customs/prediction_request_rag/prediction_request_rag.py]: changed accordingly

**Review by jmoreira-valory — APPROVED** [review]

**Pull request merged**